src/HubSpot/Companies/HubSpotCompanyConnector.cs

```diff
@@ -47,26 +47,24 @@ public async Task<TCompany> SaveAsync<TCompany>(TCompany company) where TCompany
                 throw new ArgumentNullException(nameof(company));
             }
 
-            var modifiedProperties = (from property in _typeManager.GetModifiedProperties(company)
-                                      select new ValuedPropertyV2(property.name, property.value)).ToArray();
+            var customProperties = (from property in _typeManager.GetPropertyData(company)
+                                      select new ValuedPropertyV2(property.PropertyName, property.Value?.ToString())).ToArray();
 
-            if (modifiedProperties.Any())
+            if (customProperties.All(cp => string.IsNullOrEmpty(cp.Value)))
             {
-                if (IsNew())
-                {
-                    var createdCompany = await _client.Companies.CreateAsync(modifiedProperties).ConfigureAwait(false);
-                    var newCompany = await _client.Companies.GetByIdAsync(createdCompany.Id).ConfigureAwait(false);
-                    return _typeManager.ConvertTo<TCompany>(newCompany);
-                }
-                else
-                {
-                    await _client.Companies.UpdateAsync(company.Id, modifiedProperties).ConfigureAwait(false);
-                    var modifiedCompany = await _client.Companies.GetByIdAsync(company.Id).ConfigureAwait(false);
-                    return _typeManager.ConvertTo<TCompany>(modifiedCompany);
-                }
+                return company;
```

Comment:
What if all properties has been "reset"? Wouldn't they all be empty then, and shouldn't that be saved?

Reply:
Good question. There is a [test that checks for persistence of empty contact](https://github.com/emgdev/hubspot-dotnet-sdk/blob/master/tests/Tests.Integrations/Contacts/HubSpotContactConnectorTests.cs#L266), so I wanted to not break that behavior, and also keep all the connectors consistent in usage. 

We could add a check here, to say that if the ID of the company / deal / contact is not the default value we save, otherwise we return. 
`if (customProperties.All(cp => string.IsNullOrEmpty(cp.Value)) && company.Id == default)`

What do you think of that?

Reply:
Yeah, I think that sounds like a good approach (but maybe using the `IsNew()` function that already exists).
We obviously don't want to add completely new empty companies/contacts etc. But if we (for some reason) were able to reset all the properties, we would probably want that saved, right?